tonic/src/status.rs

```diff
@@ -401,6 +402,24 @@ impl Status {
 
         Ok(())
     }
+
+    /// Create a new `Status` with the associated code, message, and binary details field.
+    pub fn with_raw_details(code: Code, message: impl Into<String>, details: Bytes) -> Status {
+        Status {
+            code,
+            message: message.into(),
+            details: details,
+        }
+    }
+
+    /// Create a new `Status` with the associated code, message, and protobuf details field.
+    pub fn with_details(code: Code, message: impl Into<String>, details: impl Message) -> Status {
```

Comment:
I actually believe we should not include a prost based one here. I think we can just have the one above that takes a Bytes and we remove the impl Into<String>. We can also rename the one above to `with_details`. then let the user choose if they want to encode via prost or something else.

Reply:
Agreed. Just copy pasta from the PR I inherited. I'm fairly new to Rust so I appreciate your patience. Is this `cargo check --no-default-features` CI check to enforce that there's no unnecessary coupling of external crates to the core library?